docs(recognitions): remove student id as a param

dcsiebe opened the pull request:
Remove student id as a request parameter since the recognition is
created as the current authenticated student.

dcsiebe committed:
docs(recognitions): remove student id as a param

Remove student id as a request parameter since the recognition is
created as the current authenticated student.

Lehren commented on docs/api/recognitions.md:
```diff
@@ -25,7 +25,6 @@ POST /challenges/choice/:challenge/recognitions HTTP/1.1
 Name       | Type     | Description
```
Forgot to mention it's created as the current student.

dcsiebe replied:
Done ;)

dcsiebe committed:
docs(recognitions): note current user is the owner

Note that when creating a speech recognition the current user is the
owner of the recognition.
